core/src/main/java/io/grpc/Contexts.java

```diff
@@ -130,4 +134,24 @@ public void onReady() {
       }
     }
   }
+
+  /**
+   * Returns the {@link Status} of a cancelled context or {@code null} if the context
+   * is not cancelled.
+   */
+  public static Status statusFromCancelled(Context context) {
+    Preconditions.checkNotNull(context, "context must not be null");
+
+    if (!context.isCancelled()) {
+      return null;
+    }
+    final Throwable cancellationCause = context.cancellationCause();
+    if (cancellationCause instanceof StatusException) {
```

Comment:
Could be StatusRuntimeException too. Probably better to do something like

if (cancellationCause instanceof TimeoutException) {
  return Status.DEADLINE_EXCEEDED.withDescription(cancellationCause.getMessage());
} else if (cancellationCause == null) {
 return Status.CANCELLED;
} else {
 // Could check for UNKNOWN here and substitute in CANCELLED but probably overkill
 return Status.fromThrowable(cancellationCause);
}